php/WC_Gateway_Blockonomics.php

```diff
@@ -195,7 +196,7 @@ public function check_blockonomics_callback()
                     $minutes = (time() - $timestamp)/60;
                     $wc_order->add_order_note(__("Warning: Payment arrived after $minutes minutes. Received BTC may not match current bitcoin price", 'blockonomics-bitcoin-payments'));
                 }
-                elseif ($status == 2) {
+                elseif ($status >= $network_confirmations) {
```

Comment:
We should avoid duplicate execution of this code. It may execute on status 0 and then again on status 1 and 2, which will cause multiple unnecessary payment notes.

Reply:
Fixed with commit 145b3682e415450e612e3dff6efe184dbf0f9d48